### PR TITLE
[CSV Export] Implement exportService

### PR DESCRIPTION
**Pull request by VWoeltjen**
Implements CSV exportService, #649 

Summary of changes:

* Add `exportService`, from bundle `platform/exporters`
* Add `example/export` as an example of its usage
* Update bower dependencies and require paths

Note that this was branched from another topic branch (see #642) in order to utilize bower-based dependency management approach

Still todo:

- [x] Update licenses
- [x] Verify that diff looks good after #642 is merged
- [x] Investigate CircleCI failure

### CODE REVIEW

**Comment by VWoeltjen** [comment]
Hm, CircleCI is failing on tests involving Blob because it is still using PhantomJS 1.9.8 for some reason (instead of 2.1.0 as specified in `package.json`). Guessing this is a globally-installed version of Phantom there? Not sure why this is being preferred over the version installed as our dependency

**Comment by larkin** [comment]
@VWoeltjen I think karma does not use the phantomjs version specified in package.json; I'm not sure what it is using.

hrm:
https://github.com/karma-runner/karma-phantomjs-launcher/blob/master/index.js#L17
https://www.npmjs.com/package/phantomjs-prebuilt

**Comment by larkin** [comment]
@VWoeltjen looks like we should update to new karma-phantom-launcher v1.0.

**Comment by VWoeltjen** [comment]
@larkin Thanks, upgrading to 1.0.0 did the trick (looks like I should have done this when switching to `phantomjs-prebuilt`, 750a5648d2df805d111a9114741fda11a68f6f3d)

**Comment by akhenry** [comment]
Don't have any other PR's for this release, so unless @larkin you had a particular attachment to this issue I'll do the review. 

@VWoeltjen It looks like there's still some work to do on this?

**Comment by VWoeltjen** [comment]
@akhenry only remaining dependency here is merge of #642

This depends on some of the changes from that branch (specifically, it uses bower to get front-end dependencies). As such, changes made based on review _there_ may effect things here (and merging this branch now would circumvent review of #642.)

**Comment by VWoeltjen** [comment]
#642 has been merged!

### Author Checklist

1. Changes address original issue? Y
2. Unit tests included and/or updated with changes? Y
3. Command line build passes? Y
4. Changes have been smoke-tested? Y

**Comment by VWoeltjen** [comment]
Merged latest from master in order to get a clean diff here (for whatever reason GitHub was still showing all the changes from #642, even after it had been merged)

File: example/export/bundle.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define*/
+
+define(['legacyRegistry'], function (legacyRegistry) {
+    "use strict";
+
+    function ExportTelemetryAsCSVAction(exportService, context) {
```
Comment (akhenry) [inline example/export/bundle.js]: This is an interesting departure from convention. Any reason why this action isn't defined in its own source file? Might make it difficult to find in the source. Also, although its brevity and simplicity in its current form don't warrant testing, defining it in the bundle will make it difficult to test or to extend in future.

File: platform/exporters/ExportService.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+ * licenses. See the Open Source Licenses file (LICENSES.md) included with
+ * this source code distribution or the Licensing information page available
+ * at runtime from the About dialog for additional information.
+ *****************************************************************************/
+/*global define,Blob*/
+
+define(['csv'], function (CSV) {
+
+    /**
+     * The `exportService` provides
```
Comment (akhenry) [inline platform/exporters/ExportService.js]: incomplete comment

File: example/export/bundle.js (hunk 1 above)
Comment (akhenry) [inline example/export/bundle.js]: Oh, because it's an example. Never mind

Comment (akhenry) [inline example/export/bundle.js]: Actually no, I take that back, examples should be code exemplars, and we don't want to encourage people to use this convention. In my opinion :)

File: platform/exporters/ExportService.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+     * @param {function} saveAs a function that takes a Blob and a file name
```
Comment (akhenry) [inline platform/exporters/ExportService.js]: Could we document this with an `@callback` just to be clear about the parameters for the saveAs function and their order.

**Comment by akhenry** [comment]
Looks good, couple of really minor points in the comments.

File: example/export/bundle.js (hunk 1 above)
Comment (VWoeltjen) [inline example/export/bundle.js]: That's a good point. I'd been thinking that this simplification was nice for an example, and in some ways it is (you can see the whole feature in one file), but it's better to be consistent about the usage we want to encourage. Will split this up.

Comment (larkin) [inline example/export/bundle.js]: I think it's really nice to see this in one file.  It greatly simplifies understanding.  Yes, you should break functionality into multiple files, but different projects can define their own module structure.  A single file is an incredibly easy to understand "getting started" guide and is very valuable.  

We're not trying to demonstrate how to properly package and separate code-- we're trying to demonstrate how to use a feature.

Comment (VWoeltjen) [inline example/export/bundle.js]: @larkin - that had been my original thinking, and that's valid for certain audiences. But in practice, I don't think we can really expect that a reader will understand the scope of the example the same way we do. (Put another way, I think this makes sense in a "Getting Started" guide, but not in a "here are various examples of plugins that use various features" context.)

**Comment by VWoeltjen** [comment]
Addressed points from code review; reassigning for merge (once we're past code freeze)

**Comment by akhenry** [comment]
Subsequent commits to master created merge conflict which I've taken the liberty of resolving.

**Comment by akhenry** [comment]
### Reviewer Checklist

1. Changes appear to address issue? __Y__
2. Appropriate unit tests included? __Y__
3. Code style and in-line documentation are appropriate? __Y__
4. Commit messages meet standards? __Y__

**Pull request merged**